ast/file_info.go

```diff
@@ -598,6 +597,46 @@ func (pos SourcePos) String() string {
 	return fmt.Sprintf("%s:%d:%d", pos.Filename, pos.Line, pos.Col)
 }
 
+func (pos SourcePos) AsSpan() SourceSpan {
+	return emptySpan{Pos: pos}
+}
+
+type emptySpan struct {
+	Pos SourcePos
+}
+
+func (e emptySpan) Start() SourcePos {
+	return e.Pos
+}
+
+func (e emptySpan) End() SourcePos {
+	return e.Pos
+}
```

Comment:
I would omit this. Just let the caller use `ast.NewSpan(pos, pos)`.

Reply:
Looks like the main callers in this repo are for an unknown position. So you might also add a `UnknownSpan` function, next to `UnknownPos`. If you wanted to be more efficient than using `NewSpan`, you could define a `type unknownSpan string`, whose underlying value is the file name.

Reply:
done